elastic/kibana
# [Infrastructure UI] Hosts technical preview discovery

Closes #148235

## Summary

This PR adds an option for the user to enable the hosts view and makes the hosts navigation link always visible. 

### Code Review
I know that this PR looks big (it actually is 😅 ) and not "fun" to review but I will try to make it easier by splitting and pointing out the different parts here: 
- [Change in the Navigation](https://github.com/elastic/kibana/pull/149351/files#diff-1f9338fabab6f332f6b4b11dbcec00f10b89097720c94b6087c53b186659c976): Contains also a [component](https://github.com/elastic/kibana/pull/149351/files#diff-9f8d59f3f98855c2cc6c4bd8ad93f69ad06181809a8769121fc1a3714c848f5e) to show the icon next to the name
- [Hosts view navigation link enabled by default](https://github.com/elastic/kibana/pull/149351/files#diff-f66e54f3274b7dd4c72b54011c1b4a1d2e4d91aa47341e14b048377437574655)
- [The Hosts View Landing page](https://github.com/elastic/kibana/pull/149351/files#diff-c3aa95031634008b7384dd670ca12c8266bead5f2cd461fdc6070fe84ae9e4e4) : Using the [EnableHostViewPage](https://github.com/elastic/kibana/pull/149351/files#diff-a0704adf74ea9605858160dfbe6ee04021b8161a63a54852009847fb5caeb5e8) component passing different `actions` based on the use case. 
- [Link on the inventory page](https://github.com/elastic/kibana/pull/149351/files#diff-181e8eeab11591c40565d6636c1b0fc6099bc555af71efe612d1da3f406d7101)
- [Functional Tests](https://github.com/elastic/kibana/pull/149351/files#diff-cdf2ef63d7f0db051d569b09fbd2ba39eb852d3d75bf3b29d85e57de2da3af35)
### Flow
#### How to navigate to Hosts View
- The link from the inventory page which will navigate the user to a separate page where he will have the option to enable hosts view (if the user has permission to enable it otherwise he will see a message to contact his administrator to do it for him). In order to enable the hosts view the user should have permission to modify advanced settings. 
-  The navigation link will be always visible. After enabling the host view the user will see the hosts view page
(links highlighted in green): <img width="2055" alt="image" src="https://user-images.githubusercontent.com/14139027/214123141-e3d8a156-839f-4dbf-ae6f-b564eccb16f7.png"> 

#### After navigating to host view
- When hosts view is disabled

| Admin User  | User with read-only permission |
| ------------ | ------------------------------- |
| ![image](https://user-images.githubusercontent.com/14139027/214124509-3286fd62-7e17-409a-8407-19499485b704.png) | ![image](https://user-images.githubusercontent.com/14139027/214123646-68540663-1525-4998-babd-172999a546a5.png) |

- When hosts view is enabled (Both Admin User  and User with read-only permission ) <img width="1713" alt="image" src="https://user-images.githubusercontent.com/14139027/214125673-8976047c-7653-4345-bfee-41413f080e05.png">

## Testing

- Remove the override (if any) in the`kibana.yml` for `observability:enableInfrastructureHostsView` 
- Use `elastic` user (or user with permission to modify advanced settings) to enable the host view (navigating from the inventory page or from the nav menu)
- to disable the host view you should set `observability:enableInfrastructureHostsView` to `false` in Stack Management > Advanced Settings: <img width="1724" alt="image" src="https://user-images.githubusercontent.com/14139027/214127034-fc245d12-0e56-4aa3-9b35-bdfeb7760960.png">
- create a new role and set the `Management` permissions to `none` and observability permissions to `read`:  <img width="1726" alt="image" src="https://user-images.githubusercontent.com/14139027/214127744-27e3014f-c181-4cd0-9b91-8868580a909b.png">
- Assign this role to a user 
- Open hosts view (for both cases `observability:enableInfrastructureHostsView` set to `false`/`true`)

❓ Testing telemetry
